features/lorawan/LoRaWANStack.cpp

```diff
@@ -121,127 +115,87 @@ LoRaWANStack& LoRaWANStack::get_lorawan_stack()
     return _lw_stack;
 }
 
-radio_events_t *LoRaWANStack::bind_radio_driver(LoRaRadio& radio)
+void LoRaWANStack::bind_radio_driver(LoRaRadio& radio)
 {
-    // Store pointer to callback routines inside MAC layer (non-IRQ safe)
-    _mac_handlers = _loramac.get_phy_event_handlers();
-    //  passes the reference to radio driver down to PHY layer
-    _lora_phy.set_radio_instance(radio);
-    return _mac_handlers;
+    _loramac.bind_radio_driver(radio);
+}
+
+lorawan_status_t LoRaWANStack::connect()
+{
+    // connection attempt without parameters.
+    // System tries to look for configuration in mbed_lib.json that can be
+    // overridden by mbed_app.json. However, if none of the json files are
+    // available (highly unlikely), we still fallback to some default parameters.
+    // Check lorawan_data_structure for fallback defaults.
+
+    lorawan_connect_t connection_params;
+
+    //TODO: LoRaWANStack don't need to know these values, move to LoRaMac (or below)
+#if (1 == MBED_CONF_LORA_OVER_THE_AIR_ACTIVATION)
```

Comment:
Just `#if MBED_CONF_LORA_OVER_THE_AIR_ACTIVATION`, surely?

You're not doing `#if (1 == defined(XXX))`...